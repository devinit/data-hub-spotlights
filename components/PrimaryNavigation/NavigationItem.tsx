import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

export interface NavigationItemProps {
  title: string;
  active: boolean;
  url: string;
}

const NavigationItem: FunctionComponent<NavigationItemProps> = ({ active, title, url }) => {
  return (
    <li className={classNames('navigation-primary__item', { 'navigation-primary__item--active': active })}>
      <a href={url}>{title}</a>
    </li>
  );
};

export { NavigationItem };
