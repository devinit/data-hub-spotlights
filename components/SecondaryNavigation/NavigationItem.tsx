import React, { FunctionComponent } from 'react';
import classNames from 'classnames';

export interface NavigationItemProps {
  title: string;
  active: boolean;
  url: string;
}

const NavigationItem: FunctionComponent<NavigationItemProps> = ({ active, title, url }) => {
  return (
    <li className={classNames('navigation-secondary__item', { 'navigation-secondary__item--active': active })}>
      <a href={url}>{title}</a>
    </li>
  );
};

export { NavigationItem };
