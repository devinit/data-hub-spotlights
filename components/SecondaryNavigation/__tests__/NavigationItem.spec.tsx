/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { NavigationItem, NavigationItemProps } from '../NavigationItem';

const item: NavigationItemProps = {
  title: 'Spotlight on Uganda',
  url: '/spotlight/spotlight-uganda',
  active: false,
};

describe('NavigationItem', () => {
  test('renders the NavigationItem correctly', () => {
    const renderer = TestRenderer.create(<NavigationItem {...item} />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with the class navigation-secondary__item--active when active', () => {
    item.active = true;
    const { container } = render(<NavigationItem {...item} />);

    expect(container.firstChild).toHaveClass('navigation-secondary__item--active');
  });
});
