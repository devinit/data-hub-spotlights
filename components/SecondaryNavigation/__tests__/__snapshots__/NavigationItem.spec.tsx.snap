// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders the NavigationItem correctly 1`] = `
<li
  className="navigation-secondary__item"
>
  <a
    href="/spotlight/spotlight-uganda"
  >
    Spotlight on Uganda
  </a>
</li>
`;
